io-sim/src/Control/Monad/IOSimPOR/Internal.hs

```diff
@@ -336,7 +391,62 @@ schedule thread@Thread{
         return (SimPORTrace time tid tstep tlbl (EventThrow e) $
                 SimPORTrace time tid tstep tlbl (EventMask maskst') trace)
 
-      Left isMain
+      -- Found a CatchFrame with a ThreadDelayFrame so we need to unregister
+      -- the threadDelay timer
+      (Right ( thread'@Thread { threadMasking = maskst' }
+             , Nothing) , Just tdid) -> do
+        let thread'' = stepThread thread'
+            control' = advanceControl (threadStepId thread') control
+            races'   = updateRacesInSimState thread' simstate
```

Comment:
I have the same question, but if you look a few lines above (or in the diff before I changed stuff) when we found a suitable exception handler we call `updateRacesInSimState`. So I figured this particular case and the ones below should behave in the same way since in a nutshell we are dealing with particular `CatchFrame`s (they are not exactly `CatchFrame`s in the sense that one called catch somewhere in the code, but they are implicit catch call)

Reply:
I don't think this is required anymore

Reply:
I would still like to hear back from John, but we don't need to be blocked on merging.